### DISCUSSION TITLE
Confused About Lora

### DISCUSSION THREAD

**Discussion by JimmysGrubhub** [d8ahazard/sd_dreambooth_extension#561] [Q&A]
I've waited a few days to post this hoping there would be more in-depth discussion but haven't seen much detail
Can someone give a rundown on Lora without getting too technical? 
Is it comparable to regular dreambooth and if so what needs to be done to achieve parity?
What are the considerations that need to be taken in comparison to regular existing db guides in terms of the two learning rates / steps / etc?

**Comment by TypicalPlay** [comment]
Same here.

**Comment by ghost** [comment]
Can't answer exactly what you ask here, because i simply can't train DB on 8 gigs to really compare. But i guess it is comparable? It does learn new words, data about tokens and such, just doesn't affect model, but saves what is different as separate file instead, which you then applying to model.

I did train multiple styles with it, and even did very light fine-tune based on cherrypicked results from base model.
From what i see, people like LORA more.

You need higher learning rate with LORA, for now just remove 2 zeros from it. Steps - i like 12-16k for ~100 image datasets.

Rest i just followed DB guides in this repo tbh. So, it is really DB but different i guess?

**Comment by mediocreatmybest** [comment]
My "basic" understanding of how it works is: 
(This doesn't mean I understand the math or the technical work on it, also PLEASE, correct my understanding if I'm totally wrong on any accounts anyone, I am happy to be told I'm wrong.)

From one of the github posts:
- **Fine-tune Stable diffusion models twice as faster than dreambooth method, by Low-rank Adaptation**
- _Low-rank Adaptation is basically using a different training method and doesn't train the full model and produces smaller delta like files. https://arxiv.org/abs/2106.09685 <-- This is a good write up._
- **Get insanely small end result (3MB for just unet, 6MB for both unet + clip), easy to share and download.**
- _The LORA trained data is pretty small, currently it is merged into the main model to be useful, hopefully they make an interface that allows loading it separately, similar to hypernetworks or textual inversion files._  
- **Easy to use, compatible with diffusers.**
- _It works with stable diffusion, just needs some work arounds currently._ 
- **Sometimes even better performance than full fine-tuning (but left as future work for extensive comparisons)**
- _I think this needs to be demonstrated still, including settings and compare to a full fine-tuned model with the same dataset._ 
- **Merge checkpoints + Build recipes by merging LoRAs together**
- _Yep, you can merge a check point with LoRA, I haven't been able to get this to work with Stable Diffusion 2.x though._ 
- **Fine-tune both CLIP & Unet to gain better results.**
- _I think this needs to be demonstrated still, including settings and compare to a full fine-tuned model with the same dataset._ 

This is my "basic" understanding anyway. 
Currently I can't get it working with Stable Diffusion 2.x

**Comment by birb3D** [comment]
From reddit on how to get a trained LORA to work with SD:

> "Create model" with the "source checkpoint" set to Stable Diffusion 1.5 ckpt.
> That model will appear on the left in the "model" dropdown.
> Now select your Lora model in the "Lora Model" Dropdown.
> (If it doesn't exist, put your Lora PT file here: Automatic1111\stable-diffusion-webui\models\lora)
> Name the model under "Custom Model Name"
> Press "Generate Ckpt".
> DONE.
> In the top left, press the purple arrow refresh button to view your newly created checkpoint!
